src/motor-operations.h

```diff
@@ -64,22 +69,30 @@ class MotorOperations {  // Rename SegmentQueue ?
 
   // Wait, until all elements in the ring-buffer are consumed.
   virtual void WaitQueueEmpty() = 0;
+
+  // Get a reference of a FixedArray containing the absolute motors
```

Comment:
What I mean: instead of returning a value, just have the user pass a pointer into the function
```
  GetCurrentMotorPos(MotorCoordSystem *here_I_want_it);
```
... and have the function fill whereever that is pointing to.

Returning a reference or pointer to some internal thing can be problematic as you don't have control over when the caller accesses the value, and if it is something that is changed in parallel; so this is why you returned a value in the first place.